### DISCUSSION TITLE
chat before session/pixels/logon

### DISCUSSION THREAD

**Discussion by aabittner** [rustdesk/rustdesk-server-pro#901] [Feature Request]
coming from alternative remote support and remote access systems, i liked the idea that some of those (e.g. anydesk) do suport a chatting with a remote target before actually entering a session. so that on quick support situations or as a security measure and proof you could talk to your remote person and telling them who you actually are and kind of proving before they might accidentally click on accept and allow everybody passing by to connect to their machine.

please implement a way to always chat first or allow the chat functionality right before one sends over usernames/passwords and even before the remote user is directly presented with an accept popup etc.

hopefully you get the idea. thank you very much.

**Comment by stSaHiB** [comment]
I'm also missing the possibility to chat, without starting a connection. Teamviewer can do that.
I often got situations like "can u solve the problem, when you got time?" or "solve it after my meeting"
So I only got 2 possibilities: make a phone call to check if now is a good time for solving the issue - that takes time and interrups the customers workflow alot. 
2nd:  just connect to the computer (permament password) but that hurts the persons privacy, because Im not knowing what the person is doing right now.

Chat gives the possibility to ask "can I fix it now?"
For me an important feature that I am missing at least 1-2x a week and I am not even a full time supporter ;-)

**Comment by rustdesk** [comment]
Could you confirm if "connect with message" (https://github.com/rustdesk/rustdesk-server-pro/discussions/862#discussioncomment-15681370) is ok for you?

**Reply by stSaHiB** [reply]
> Could you confirm if "connect with message" ([rustdesk/rustdesk-server-pro#862 (reply in thread)](https://github.com/rustdesk/rustdesk-server-pro/discussions/862#discussioncomment-15681370)) is ok for you?

nah, that will still hurt the others users privacy. imagine you are on the job, watching "cat" videos, or just having an open document with content you dont wanna share with your tech support. If I connect with a message, it's already too late - tech support seen things he is not supposed to see. So just the simple chat, like it is already build in, when connected, would be nice.

**_Can i connect?_**
_sure, but gimme a moment to finnish some stuff_
_OK, done, feel free to connect_

**Reply by rustdesk** [reply]
You did not understand "connect with a message" yet. It send message first, you can chat, the screen sharing won't be started before your consent.

**Reply by StealUrKill** [reply]
> You did not understand "connect with a message" yet. It send message first, you can chat, the screen sharing won't be started before your consent.

I believe so. 

Controlling end initiate remote chat to client and chat window pops up on far end with chat. Like instant messaging but in rustdesk.

**Reply by stSaHiB** [reply]
> You did not understand "connect with a message" yet.  

You're right, seems like I've missed a part of that discussion - sorry.
I dont see, why you want to bundle the connect and the chat in one button. Chat alone, something like any instant messanger, would be better for my usecase. Although we are not using a public server, I Like the idea, that you can only chat with people you already had a connection with, as a spam protection.
I am eager to see that feature in one of the upcoming releases. Thx 4 ur support!

**Comment by retechp** [comment]
+1 this feature
